### PR TITLE
Add support to accept "skip_node_allocation" flag as kwargs

### PR DESCRIPTION
**Pull request by sonupillai**
This adds the ability to skip node allocation for a service. With this, another layer of abstraction
can be written over ductape services when the nodes and services are already known and are running and we just need to run some tests on the running services.

Ran all the unit tests using tox and got a clean run.

### CODE REVIEW

File: ducktape/command_line/parse_args.py [hunk 1]
```diff
@@ -70,6 +70,8 @@ def create_ducktape_parser():
                         help="URL of a JSON report file containing stats from a previous test run. If specified, "
                              "this will be used when creating subsets of tests to divide evenly by total run time "
                              "instead of by number of tests.")
+    parser.add_argument("--allocate_nodes", action="store_true", help="Use this flag to skip allocating nodes for "
```
Comment (scott-hendricks) [inline ducktape/command_line/parse_args.py]: convention is to have this be `--allocate-nodes`

File: ducktape/services/service.py [hunk 2]
```diff
@@ -76,7 +76,9 @@ def __init__(self, context, num_nodes=None, cluster_spec=None, *args, **kwargs):
         self.context = context
 
         self.nodes = []
-        self.allocate_nodes()
+        self.skip_nodes_allocation = kwargs.get("skip_nodes_allocation", False)
```
Comment (scott-hendricks) [inline ducktape/services/service.py]: how does this kwarg get passed, if there's no mechanism to bring the command line argument into this function?

Comment (sonupillai) [reply ducktape/services/service.py]: Type in parser. Fixed. Thank you!

File: ducktape/command_line/parse_args.py [hunk 3]
```diff
@@ -70,6 +70,8 @@ def create_ducktape_parser():
                         help="URL of a JSON report file containing stats from a previous test run. If specified, "
                              "this will be used when creating subsets of tests to divide evenly by total run time "
                              "instead of by number of tests.")
+    parser.add_argument("--skip_nodes_allocation", action="store_true", help="Use this flag to skip allocating nodes for "
```
Comment (scott-hendricks) [inline ducktape/command_line/parse_args.py]: convention is to use `--skip-nodes-allocation`

**Pull request merged**